lib/IRGen/IRGen.cpp

```diff
@@ -383,8 +383,11 @@ static bool needsRecompile(StringRef OutputFilename, ArrayRef<uint8_t> HashData,
 
   // Search for the section which holds the hash.
   for (auto &Section : ObjectFile->sections()) {
-    StringRef SectionName;
-    Section.getName(SectionName);
+    llvm::Expected<StringRef> SectionNameOrErr = Section.getName();
+    if (auto Err = SectionNameOrErr.takeError())
```

Comment:
No need to takeError here, you can use operator bool () and that will fulfill assertIsChecked when the destructor is called and still clean up there error if there is one.

Same for the rest.